src/query_plan/query_planning_traversal.rs

```diff
@@ -86,3 +89,108 @@ struct BestQueryPlanInfo {
+                .supergraph_schema
+                .schema()
+                .root_operation(self.root_kind.into())
+                .cloned()
+                // A root operation type has to be an object type
+                .map(|type_name| ObjectTypeDefinitionPosition { type_name }.into())
+        } else {
+            None
+        };
+        FetchDependencyGraph::new(
+            self.parameters.supergraph_schema.clone(),
+            self.parameters.federated_query_graph.clone(),
+            root_type,
+            self.starting_id_generation,
+        )
+    }
+
+    fn updated_dependency_graph(
+        &self,
+        dependency_graph: &mut FetchDependencyGraph,
+        path_tree: &OpPathTree,
+    ) -> Result<(), FederationError> {
+        let is_root_path_tree = matches!(
+            path_tree.graph.node_weight(path_tree.node)?.type_,
+            QueryGraphNodeType::SchemaType(_)
+        );
+        if is_root_path_tree {
+            // The root of the pathTree is one of the "fake" root of the subgraphs graph,
+            // which belongs to no subgraph but points to each ones.
+            // So we "unpack" the first level of the tree to find out our top level groups
+            // (and initialize our stack).
+            // Note that we can safely ignore the triggers of that first level
+            // as it will all be free transition, and we know we cannot have conditions.
+            for child in &path_tree.childs {
+                let edge = child.edge.expect("The root edge should not be None");
+                let (_source_node, target_node) = path_tree.graph.edge_endpoints(edge)?;
+                let target_node = path_tree.graph.node_weight(target_node)?;
+                let subgraph_name = &target_node.source;
+                let subgraph_schema = self
+                    .parameters
+                    .federated_query_graph
+                    .schema_by_source(subgraph_name)?;
+                let root_type = match &target_node.type_ {
+                    QueryGraphNodeType::SchemaType(OutputTypeDefinitionPosition::Object(
+                        object,
+                    )) => object.clone().into(),
+                    ty => {
+                        return Err(FederationError::internal(format!(
+                            "expected an object type for the root of a subgraph, found {ty}"
+                        )))
+                    }
+                };
+                let fetch_dependency_node = dependency_graph.get_or_create_root_node(
+                    subgraph_name,
+                    subgraph_schema,
+                    self.root_kind,
+                    root_type,
+                );
+                compute_nodes_for_tree(
+                    dependency_graph,
+                    &child.tree,
+                    fetch_dependency_node,
+                    Default::default(),
+                    Default::default(),
+                    Conditions::TRUE,
```

Comment:
Instead of `Conditions::TRUE`, you'll want an `OpGraphPathContext` here with an empty vector for conditionals (could use `default()` here as well).